### DISCUSSION TITLE
2021 Summer Virtual Internship Program

### DISCUSSION THREAD

**Discussion by ChristopherA** [BlockchainCommons/Community#32] [Q&A]
## Overview

This summer, Blockchain Commons is again offering a *Virtual Internship Program* as part of our mission to support open infrastructure for blockchain, Bitcoin, security & privacy, and digital civil rights.

This is a great opportunity to learn valuable real-world open-source skills. The internship is open not only to software and hardware developers, but  to people with other skills such as pre-law, library science, technical writing, etc. who want to enter this field. We are also open to corporate [“Externship”](https://en.wikipedia.org/wiki/Externship/) / ["Secondment"](https://en.wikipedia.org/wiki/Secondment).

If you are interested in applying to become a Blockchain Common intern, the details are in this thread. Also post if you have any questions that are not fully answered here.

## History 

Blockchain Commons hosted a successful Virtual Internship Program in Summer 2020 during the Covid-19 crisis, with Interns from around the world and in many time zones. This internship was led by Christopher Allen @ChristopherA who has also mentored a number of Bitcoin and security-related interns over the last two decades.

Two articles about 2020 Virtual Intership Program were written up in the press, one at the beginning of the program, one at the end:

* [2020/07/03 Decrypt: How a Bitcoin dev hub is building a future for free software](https://decrypt.co/34555/how-bitcoin-dev-hub-building-future-free-software)
* [2020/09/03 Coindesk: Blockchain Commons Internship Introduces New Developers to Open Source](https://www.coindesk.com/blockchain-commons-interns-open-source-developers)

## Requirements

Interns are expected to commit to at least 40 hours of professional effort toward their internship project over the course of 2-3 months. We do offer a small honorarium over the course of the program as the Interns meet 3 milestones and are compensated a minimum of US$564 in Bitcoin as part of an [Intership Agreement](https://github.com/BlockchainCommons/Open-Development/tree/master/internship-contract) that parallels how we contract professional development. All learning hours and any project hours in excess of 40 hours are at the Intern's own discretion as normal participation in an open-source project. Due to our grant, we may be able to make some additional honorarium payments to interns who contribute more time and effort.

Interns are also expected to actively participate in a private Signal group with fellow interns and members of the @BlockchainCommons community, as well as weekly open "intern office hours" video chats with open questions and frequent interviews with guests from the industry. In addition, each intern is offered 2 personal "mentorship" calls with @ChristopherA to discuss their education and career plans for the future.

We are primarily focused on creating a learning experience for software developers, but we also welcome applications from the hardware community and those outside of the developer community, as not all projects are code. In particular, we are seeking library science students to help us organize our links, references, and work; education specialists to work on tutorials; pre-law students to help with our advocacy in the creation of new laws in places like Wyoming where our advice has been well-respected and is turning into law; and researchers of any type to support the creation of engagement models.

At whatever level of development skill you do have (or not) we hope that you'll at least have worked through the first chapters of the [Learning Bitcoin from the Command Line](https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line) course and have gotten to as far as you can. Command line isn't easy, but it isn't hard either! We also ask that have tested out at least one of our open-source projects, such as one of the [Gordian betas](https://github.com/BlockchainCommons/Gordian).

## Internship Cohorts in 2021

We hope to have a cohort of 5-7 Interns who will work together on projects. 

We expect to mold projects to fit the passions of interns, provided they also support the goals of Blockchain Commons. The following are projects we are currently considering:

* Expansion of our Gordian system, which uses torgaps and airgaps to create an architecture for Bitcoin usage that supports agency, autonomy, interoperability, and security — putting users in control of their finances, while ensuring their privacy and non-correlation.
* Bundling of existing Blockchain Commons software such as Gordian into Debian packages that can be automatically installed onto security-focused systems like Tails OS.
* Creation of user documents to make those tools accessible to even non-technological users.
* Interviewing of activists to support the development of new engagement models similar to [Rebooting Web of Trust's Amira](https://w3c-ccg.github.io/amira/), allowing everyone to better understand the use cases for activism.

### A Focus on Activism

We expect at least a third of our work in Summer 2021 to be focused on supporting activists. This has been a long-held goal at Blockchain Commons, where we're working to protect people, not just money. A grant from the [Human Rights Foundation](https://hrf.org/) is putting this work front and center this summer. Many of the above projects that we are considering are purposefully focused on support for activists.

## Initial Application

Email [Christopher Allen @ChristopherA](mailto:ChristopherA@LifeWithAlacrity.com)  with your CV, along with a cover letter that details highlights of some of the things you like to work on with Blockchain Commons, and what you’d like to get out of an internship with us. Please include what hardware platform, smart phone, and development environments you are most comfortable with.

The deadline is April 30, 2021.

## Followup Questions

After initial application, you may be asked for additional details from the following list of questions: 

(Note that these questions are focused toward developers; we do not expect these skills for Law, Library Science, and Education interns, and would instead have different followup questions.)

* What is your primary development environment and what hardware do you run it on? What kind of smartphone do you use and which version of Android or iOS?

* As a software developer, what languages and tools do you LIKE, vs ones you know and can use.

* A key part of being successful as an intern is picking and scoping a new feature in an existing open-source project or doing so for a new small project that you will lead, that is relevant to Blockchain Commons and our advocacy and that can be completed in three months. You'll collaborate with the other interns with their projects, but there should be one where you are the lead. Given that, any early thoughts on what kinds of existing Blockchain Commons projects that might want to add a feature to? Or some small new project that fits into our advocacy?

* You clearly care about and have some advocacy around Bitcoin, blockchain, and related technologies, but what aspect of it drives you to do so?

* Software developers are whole people, not just coding machines. What professional skills and talents do you bring to the table that are not about code?

* What specifically would you like to learn or get out of a 3-month internship with Blockchain Commons? What appealed to you that you are applying for an internship with us?

## Corporate 

We are open to working with organizations to provide [“Externship”](https://en.wikipedia.org/wiki/Externship/) / ["Secondment"](https://en.wikipedia.org/wiki/Secondment) opportunities to help increase professional skills of their staff in open-source secure development best practices. For a fee we can also offer "pair programming"-like opportunities with specific professional open-source developers. Contact Christopher Allen for more information.

## Sponsoring our Internship Program

Thank you to [HRF](https://hrf.org/) for an initial grant for the Summer 2021 interns!

To join in the support of our Virtual Internship Program, and other educational and career programs at Blockchain Commons, please consider becoming a [Sponsor](https://github.com/sponsors/BlockchainCommons) here at GitHub. However, you can alternatively support Blockchain Commons and our vision of the open web with Bitcoin via our [BTCPay contribution](https://btcpay.blockchaincommons.com) page.

We look forward to collaborating with you!

**Comment by kouloumos** [comment]
Are there plans for a 2022 Internship Program?

**Comment by ChristopherA** [comment]
Yes. we'll be posting details soon, but basically, it will start light in May/June, with 10 weeks of more intensive work in July August.
